src/gemini_supply/agent.py

```diff
@@ -269,16 +268,17 @@ async def get_model_response(
           message = (
             f"Generating content failed on attempt {attempt + 1}. Retrying in {delay} seconds...\n"
           )
-          termcolor.cprint(
-            self._with_agent_prefix(message),
-            color="yellow",
-          )
+          if self._logger and self._agent_label:
```

Comment:
Let's make Logger non-nullable in this class, and have it default to a NoopActivityLogger.

Actually, can we do that everywhere? The fewer nulls the better.